### PR TITLE
fix #7548 by updating to v142 toolset (again)

### LINKED ISSUE #7548: Upgrade IIS C++ projects to build with platformtoolset v142 (VS2019)
Depends on https://github.com/aspnet/AspNetCore/pull/7005

Let's update the C++ projects for ANCM to build with the Visual Studio 2019 toolset. This should be a matter of updating the .vcxproj and reacting to any warnings or errors the new toolset finds.
```
<PlatformToolset>v142</PlatformToolset>
```

### PR DESCRIPTION
**Pull request by analogrelay**
This time, it's personal.

Trying again to update to v142. We'll see how this goes ;).

We can probably also remove the v141 toolset from the vs.json files as a result of this, but I'd like to wait until we know this works to do that :). If we get a green check, we can do a second pass to clean-up (in this PR, but just after we know it works)

Before merge checklist:
* [x] Remove v141 toolset install from `korebuild.json`, `eng\scripts\vs.buildtools.json` and `eng\scripts\vs.json` files

### CODE REVIEW

**Comment by natemcmaster** [comment]
I believe this means we can also delete these two lines: https://github.com/aspnet/AspNetCore/blob/d1138f2cdeda8e04755ffec138b9514d0ffd43a8/korebuild.json#L19-L20

**Review by natemcmaster — APPROVED** [review]
If it works, :shipit:

cc @dougbu - this would unblock https://github.com/aspnet/AspNetCore-Internal/issues/2538

**Comment by anurse** [comment]
After this change, nothing should be on toolset v142 (we do have it in our VS install scripts and an unfortunate coincidence in some SVG files ;))

image

**Comment by jkotalik** [comment]
> we do have it in our VS install scripts 

Which we can remove now?

File: src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HandleWrapper.h
Comment (anurse) [inline src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HandleWrapper.h]: Being optimistic that the [compiler bug that prevented this](https://developercommunity.visualstudio.com/content/problem/33928/constexpr-failing-on-nullptr-v141-compiler-regress.html) is fixed now. We'll see :).

File: src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HandleWrapper.h
Comment (anurse) [inline src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HandleWrapper.h]: VSCode wanted to reformat all the things.

File: src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HandleWrapper.cpp
Comment (jkotalik) [inline src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HandleWrapper.cpp]: If this is fixed yeah. I think it will fail to build if this isn't fixed, so we have "test coverage"

**Review by jkotalik — APPROVED** [review]
Hoping for the best.

**Comment by anurse** [comment]
image

image

image

image

File: src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HandleWrapper.cpp
Comment (anurse) [reply src/Servers/IIS/AspNetCoreModuleV2/CommonLib/HandleWrapper.cpp]: Yep, I realized it was literally a compiler bug, so it felt safe to try :)

**Comment by anurse** [comment]
image

Success!

**Comment by anurse** [comment]
Going to push to an AzDO branch to confirm it's all good on official agents too.

**Comment by anurse** [comment]
https://dev.azure.com/dnceng/internal/_build/results?buildId=213480 is the official build I kicked off, for reference.

**Comment by natemcmaster** [comment]
🎉 nice!

**Comment by jkotalik** [comment]
@anurse just to confirm, you did verify this locally, correct?

**Comment by anurse** [comment]
`./build -BuildNative` is happy on my box. I still have the `v141` toolset on though (need it for other things). Both the test official and PR validation builds worked.

I'll leave it for @jkotalik to hit the final merge button here since I will be much less available over the next couple days and don't want to merge and bail ;). At least if @jkotalik merges he knows what to revert!

**Comment by jkotalik** [comment]
I’ll take liability.

**Pull request merged**
